src/document.c

```diff
@@ -366,7 +366,7 @@ FIELD_PREPROCESSOR(fulltextPreprocessor) {
     RSSortingVector_Put(aCtx->sv, fs->sortIdx, (void *)c, RS_SORTABLE_STR);
   }
 
-  if (FieldSpec_IsIndexable(fs)) {
+  if (FieldSpec_IsIndexable(fs) && *c != '\0') {
```

Comment:
Do you still need the `*c != '\0'`?

Reply:
Thought to leave as an optimization but since very unlikely, removed it